indexer/src/test/java/au/org/aodn/esindexer/BaseTestClass.java

```diff
@@ -187,31 +189,51 @@ public void login() {
         }
     }
 
-    public String deleteRecord(String uuid) {
+    public void deleteRecord(String... uuids) {
+        CountDownLatch latch = new CountDownLatch(1);
 
-        logger.info("Deleting GN doc {}", uuid);
-        HttpEntity<String> requestEntity = getRequestEntity(null);
+        try {
+            HttpEntity<String> requestEntity = getRequestEntity(null);
 
-        ResponseEntity<String> r = testRestTemplate
-                .exchange(
-                        getRecordUrl(uuid),
-                        HttpMethod.DELETE,
-                        requestEntity,
-                        String.class
-                );
-        logger.info("{}", r.getStatusCode());
+            // Index the item so that query yield the right result before delete
+            ResponseEntity<Void> t = testRestTemplate
+                    .exchange(
+                            getIndexUrl(),
+                            HttpMethod.PUT,
+                            requestEntity,
+                            Void.class
+                    );
+            assertEquals("Trigger index OK", HttpStatus.OK, t.getStatusCode());
+            // Cannot execute too fast, give it some wait time
+            latch.await(2, TimeUnit.SECONDS);
 
-        // Index the item so that query yield the right result
-        ResponseEntity<Void> t = testRestTemplate
-                .exchange(
-                        getIndexUrl(),
-                        HttpMethod.PUT,
-                        requestEntity,
-                        Void.class
-                );
-        assertEquals("Trigger index OK", HttpStatus.OK, t.getStatusCode());
+            for(String uuid: uuids) {
+                logger.info("Deleting GN doc {}", uuid);
+                ResponseEntity<String> r = testRestTemplate
```

Comment:
and `r` here as well , these single letter variable names growing, will get to the point `t` in this class means this, `t` in that class means that ... etc. maybe hard to read and understand in future.

Reply:
done